src/citrine/resources/data_concepts.py

```diff
@@ -677,6 +673,58 @@ def list_by_name(self, name: str, exact: bool = False,
             params=params)
         return (self.build(raw) for raw in raw_objects)
 
+    def list_by_attribute_bounds(
+            self,
+            attribute_bounds: Dict[Union[AttributeTemplate, LinkByUID], BaseBounds],
+            forward: bool = True, per_page: int = 100) -> Iterator[DataConcepts]:
+        """
+        Get all objects in the collection with attributes within certain bounds.
+
+        Results are ordered first by dataset, then by attribute value.
+
+        Currently only one attribute and one bounds on that attribute is supported, and
+        attribute type must be numeric.
+
+        Parameters
+        ----------
+        attribute_bounds: Dict[Union[AttributeTemplate, \
+        :py:class:`LinkByUID <taurus.entity.link_by_uid.LinkByUID>`], \
+        :py:class:`BaseBounds <taurus.entity.bounds.base_bounds.BaseBounds>`]
+            A dictionary from attributes to the bounds on that attribute.
+            Currently only real and integer bounds are supported.
+            Each attribute may be represented as an AttributeTemplate (PropertyTemplate,
+            ParameterTemplate, or ConditionTemplate) or as a LinkByUID,
+            but in either case there must be a uid and it must correspond to an
+            AttributeTemplate that exists in the database.
+            Only the uid is passed, so if you would like to update an attribute template you
+            must register that change to the database before you can use it to filter.
+        forward: bool
+            Set to False to reverse the order of results (i.e. return in descending order).
```

Comment:
This method allow the user to specify `forward` or `reverse`, but `list_by_tag` does not

Reply:
That's intentional -- ascending/descending order to the results of this function is actually meaningful, but in `list_by_tag`, which is ordered by UUIDS, I can think of no reason to support it. I think it's best to omit it until people think of a use case. That leaves more flexibility in the backend.

Reply:
That makes perfect sense. I actually thought that both methods returned objects in a meaningless order, and was going to advocate for removing the argument in both cases, but I agree it should be used here.